test/Test/Extensions/Parser.hs

```diff
@@ -0,0 +1,56 @@
+module Test.Extensions.Parser
+    ( parserSpec
+    ) where
+
+import Data.Text (pack)
+import Data.Text.Encoding (encodeUtf8)
+import GHC.LanguageExtensions.Type (Extension (..))
+import Test.Hspec (Arg, Expectation, Spec, SpecWith, describe, it, shouldBe)
+
+import Extensions.Parser (ExtensionsParseResult, parseSourceFile)
+
+
+parserSpec :: Spec
+parserSpec = describe "Haskell file Extensions Parser" $ do
+    itShouldParse "{-# LANGUAGE TypeApplications #-}" (Right [TypeApplications])
+    itShouldParse "{-#language TypeApplications#-}" (Right [TypeApplications])
+    itShouldParse "{-# LANGUAGE TypeApplications, LambdaCase#-}" (Right [TypeApplications, LambdaCase])
+    itShouldParse "{-# LANGUAGE   TypeApplications  , LambdaCase   #-}" (Right [TypeApplications, LambdaCase])
+    itShouldParse ( unlines
+        [ "{-# LANGUAGE TypeApplications #-}"
+        , "{-# LANGUAGE LambdaCase #-}"
+        ] )
+        (Right [TypeApplications, LambdaCase])
+    itShouldParse (unlines
+        [ "{-# LANGUAGE"
+        , " TypeApplications,"
+        , " LambdaCase  "
+        , "#-}"
+        ])
+        (Right [TypeApplications, LambdaCase])
+    itShouldParse (unlines
+        [ "{-# LANGUAGE   TypeApplications  , LambdaCase   #-}"
+        , "{- hello -}"
+        ])
+        (Right [TypeApplications, LambdaCase])
+    itShouldParse (unlines
+        [ "{-# LANGUAGE TypeApplications #-}"
+        , "{- hello -}"
+        , "{-# LANGUAGE LambdaCase #-}"
+        ])
+        (Right [TypeApplications, LambdaCase])
+    itShouldParse (unlines
+        [ "{-# LANGUAGE TypeApplications #-}"
+        , "  "
+        , "{- hello -}"
+        , "{-# LANGUAGE LambdaCase #-}"
+        , ""
+        , "hello :: IO ()"
+        ])
+        (Right [TypeApplications, LambdaCase])
+
+  where
+    itShouldParse :: String -> ExtensionsParseResult -> SpecWith (Arg Expectation)
+    itShouldParse input res = it
+        ("should parse:\n" <> unlines (map ("    " <>) $ lines input)) $
+            parseSourceFile (encodeUtf8 $ pack input) `shouldBe` res
```

Comment:
Maybe you can simplify tests a bit, if this function would take `[Extension]` and and you would just use the `Right` constructor in the function itself
```suggestion
    itShouldParse :: String -> [Extension] -> SpecWith (Arg Expectation)
    itShouldParse input res = it
        ("should parse:\n" <> unlines (map ("    " <>) $ lines input)) $
            parseSourceFile (encodeUtf8 $ pack input) `shouldBe` Right res
```